### PR TITLE
rest: add tests for read and list in readclient pkg

### PR DESCRIPTION
**Pull request by filariow**
Signed-off-by: Francesco Ilario <filario@redhat.com>

### CODE REVIEW

**Comment by sadlerap** [comment]
Also looks like it's got a merge conflict that needs to be resolved.

File: server/persistence/readclient/readclient_list_test.go [hunk 1]
```diff
@@ -29,6 +34,69 @@ var _ = Describe("List", func() {
...
+							},
+						},
+						{ObjectMeta: expectedObjectMeta},
+					}
+					return nil
+				}).
+				Times(1)
+
+				// mapper expects to be called once.
+				// It returns the list of mapped workspaces (just objectmeta)
```
[the remark this answers is not in the document]
Comment (filariow) [reply server/persistence/readclient/readclient_list_test.go:76]: this also happens below, but it's done by `go fmt` itself

File: server/persistence/readclient/readclient_read_test.go [hunk 2]
```diff
@@ -29,6 +34,50 @@ var _ = Describe("Read", func() {
...
+
+		// when
+		returnedWorkspace := restworkspacesv1alpha1.Workspace{}
+		err := rc.ReadUserWorkspace(ctx, "", "", "", &returnedWorkspace)
+
+		// then
+		Expect(err).NotTo(HaveOccurred())
+		Expect(returnedWorkspace).To(Equal(mappedWorkspace))
+		// test data is deepcopied
+		Expect(&returnedWorkspace).NotTo(BeIdenticalTo(&mappedWorkspace))
```
[the remark this answers is not in the document]
Comment (filariow) [reply server/persistence/readclient/readclient_read_test.go]: great catch! fixed in 4b9ee0872f52ca56692b75d954d09a5348ee2415

File: server/persistence/readclient/readclient_list_test.go [hunk 3]
```diff
@@ -29,6 +34,69 @@ var _ = Describe("List", func() {
...
+
+			// when
+			actualWorkspaces := restworkspacesv1alpha1.WorkspaceList{}
+			err := rc.ListUserWorkspaces(ctx, user, &actualWorkspaces, client.MatchingLabels{"whatever": "whatever"})
+
+			// then
+			Expect(err).ToNot(HaveOccurred())
+			Expect(actualWorkspaces.Items).To(HaveLen(1))
+			Expect(actualWorkspaces.Items[0].ObjectMeta).To(Equal(expectedObjectMeta))
+		})
```
[the remark this answers is not in the document]
Comment (filariow) [reply server/persistence/readclient/readclient_list_test.go:96]: good idea, let me work on this two test cases

[the remark this answers is not in the document]
Comment (filariow) [reply server/persistence/readclient/readclient_list_test.go:96]: added in ec80332cf76eb2e009de3447b263b3d01dfd697e

File: server/persistence/readclient/readclient_list_test.go [hunk 4]
```diff
@@ -29,6 +34,129 @@ var _ = Describe("List", func() {
...
+		err := rc.ListUserWorkspaces(ctx, user, &actualWorkspaces, client.MatchingLabels{"hit": "hit"})
+
+		// then
+		Expect(err).ToNot(HaveOccurred())
+		Expect(actualWorkspaces.Items).To(HaveLen(len(expectedObjectMetas)))
+		oww := make([]metav1.ObjectMeta, len(actualWorkspaces.Items))
+		for i, w := range actualWorkspaces.Items {
+			oww[i] = w.ObjectMeta
+		}
+		Expect(oww).To(BeEquivalentTo(expectedObjectMetas))
```
Comment (filariow) [inline server/persistence/readclient/readclient_list_test.go:80]: @sadlerap do you have any suggestions here?

Comment (sadlerap) [reply server/persistence/readclient/readclient_list_test.go]: I have a way of doing this without copying the `ObjectMeta` fields out, but it's longer than this.  This is more than fine for now.

**Review by sadlerap — APPROVED** [review]

**Pull request merged**